invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/CommunitiesList.js

```diff
@@ -0,0 +1,116 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2023 CERN.
+//
+// InvenioRDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import React, { Component } from "react";
+// import PropTypes from "prop-types";
+import axios from "axios";
+import { withCancel } from "react-invenio-forms";
+import { Container, Item, Button, Placeholder, Message } from "semantic-ui-react";
+import { Image } from "react-invenio-forms";
+import { CommunitiesListModal } from "./CommunitiesListModal/CommunitiesListModal";
+
+export class CommunitiesList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      communities: null,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.getCommunities();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchCommunities?.cancel();
```

Comment:
instead of using `?` operator maybe it makes sense to declare it as a class attribute?

Reply:
I'm not sure what you mean? It's already defined in here https://github.com/inveniosoftware/react-invenio-forms/blob/master/src/lib/elements/withCancel.js